controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,137 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	"github.com/openshift/ocs-operator/controllers/defaults"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ocsCephRbdMirrors struct{}
+
+func (r *StorageClusterReconciler) fetchCephRbdMirrorInstance(cephRbdMirror *cephv1.CephRBDMirror) (*cephv1.CephRBDMirror, error) {
+	existing := &cephv1.CephRBDMirror{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: cephRbdMirror.Namespace}, existing)
+	return existing, err
+}
+
+func (r *StorageClusterReconciler) deleteCephRbdMirrorInstance(cephRbdMirrors []*cephv1.CephRBDMirror, flowName string) error {
+	for _, cephRbdMirror := range cephRbdMirrors {
+		existing, err := r.fetchCephRbdMirrorInstance(cephRbdMirror)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return fmt.Errorf("%s: unable to retrieve CephRbdMirror %v: %v", flowName, cephRbdMirror.Name, err)
+			}
+			continue
+		}
+		if cephRbdMirror.GetDeletionTimestamp().IsZero() {
+			r.Log.Info(flowName+": Deleting CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			err := r.Client.Delete(context.TODO(), existing)
+			if err != nil {
+				r.Log.Info(flowName+": Retrying deletion of CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+				err := r.Client.Delete(context.TODO(), existing)
+				if err != nil {
+					r.Log.Error(err, flowName+": Failed to delete CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+					return fmt.Errorf("%s: Failed to delete CephRbdMirror %v: %v", flowName, existing.Name, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// newCephRbdMirrorInstances returns the cephRbdMirror instances that should be created
+// on first run.
+func (r *StorageClusterReconciler) newCephRbdMirrorInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephRBDMirror, error) {
+	ret := []*cephv1.CephRBDMirror{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephRbdMirror(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: cephv1.RBDMirroringSpec{
+				Count:     1,
+				Resources: defaults.DaemonResources["rbd-mirror"],
```

Comment:
https://github.com/openshift/ocs-operator/blob/b67106a5e4e56c0b8eb98d76f13e93a65f597019/controllers/defaults/utils.go#L10-L15
Use this instead. Resources can be updated from StorageCluster spec.

Reply:
ack